src/coreclr/src/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/DelegateCtorSignature.cs

```diff
@@ -88,7 +88,7 @@ public override int CompareToImpl(ISortableNode other, CompilerComparer comparer
             if (result != 0)
                 return result;
 
-            return _methodToken.CompareTo(otherNode._methodToken);
+            return _methodToken.Token.CompareTo(otherNode._methodToken.Token);
```

Comment:
Is there a guarantee that the same token here will always be accompanied by the same method? To put it yet another way, why can't we use the comparison of the complete MethodAndToken instances?

Reply:
Well, we don't have a comparison for MethodAndToken instances. And as a slight detail, my analysis of pathways into this code indicate that it will always be unique as long as the target method is also unique. I agree though, its not very obvious.

Reply:
Ah, OK, makes sense.

Reply:
Um, now that I think about it, that's a fair bit of hubris on my part that my analysis is certain to be correct now, and will remain correct in the future. I'll add the appropriate comparisons such that this is more reliable.